src/Language/Haskell/To/Elm.hs

```diff
@@ -711,7 +719,25 @@ instance (HasElmType a, KnownNat numParams, SOP.HasDatatypeInfo a, SOP.All2 (Has
                     (Expression.App "Json.Encode.string" $ Expression.String $ constructorJSONName constr)
               ]
 
-          _ -> error "Only the DataAeson.TaggedObject sumEncoding is currently supported"
+          Aeson.ObjectWithSingleField ->
```

Comment:
There's a lot of repetition with the previous case. I should perhaps push the case statement a few levels inwards.